src/screens/ENSSearchSheet.js

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Keyboard } from 'react-native';
+import { KeyboardArea } from 'react-native-keyboard-area';
+import { useDispatch } from 'react-redux';
+import dice from '../assets/dice.png';
+import TintButton from '../components/buttons/TintButton';
+import {
+  SearchInput,
+  SearchResultGradientIndicator,
+} from '../components/ens-registration';
+import { SheetActionButton, SheetActionButtonRow } from '../components/sheet';
+import { useNavigation } from '../navigation/Navigation';
+import {
+  Box,
+  Divider,
+  Heading,
+  Inline,
+  Inset,
+  Stack,
+  Text,
+} from '@rainbow-me/design-system';
+import {
+  useAccountSettings,
+  useDebounceString,
+  useENSRegistration,
+  useENSRegistrationCosts,
+  useKeyboardHeight,
+} from '@rainbow-me/hooks';
+import { ImgixImage } from '@rainbow-me/images';
+import { startRegistration } from '@rainbow-me/redux/ensRegistration';
+import Routes from '@rainbow-me/routes';
+import { colors } from '@rainbow-me/styles';
+import { normalizeENS } from '@rainbow-me/utils';
+
+export default function ENSSearchSheet() {
+  const dispatch = useDispatch();
+  const { navigate } = useNavigation();
+  const keyboardHeight = useKeyboardHeight();
+  const { accountAddress } = useAccountSettings();
+
+  const topPadding = android ? 29 : 19;
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const debouncedSearchQuery = useDebounceString(searchQuery);
+
+  const {
+    data: registrationData,
+    isIdle,
+    isRegistered,
+    isLoading,
+    isInvalid,
+    isAvailable,
+  } = useENSRegistration({
+    name: debouncedSearchQuery,
+  });
+
+  const {
+    data: registrationCostsData,
+    isSuccess: registrationCostsDataIsAvailable,
+  } = useENSRegistrationCosts({
+    duration: 1,
+    name: debouncedSearchQuery,
+    rentPrice: registrationData?.rentPrice,
+  });
+
+  const state = useMemo(() => {
+    if (isAvailable) return 'success';
+    if (isRegistered || isInvalid) return 'warning';
+    return 'rainbow';
+  }, [isAvailable, isInvalid, isRegistered]);
+
+  const handlePressContinue = useCallback(() => {
+    dispatch(startRegistration(accountAddress, `${searchQuery}.eth`));
```

Comment:
note to self: confirm `searchQuery` itself has been normalized in case it does also include `.eth` before passing it on to `startRegistration`